src/templates/ApiReference.js

```diff
@@ -1,207 +1,58 @@
-  padding: 0.25rem 0;
-  padding-left: ${(props) => (props.depth > 1 ? `${props.depth - 1}rem` : 0)};
-  transition: opacity ${CSS_TRANSITION_SPEED.default} ease-out;
-  font-weight: ${FONT_WEIGHT.normal};
-
-  ${(props) =>
-    props.isActive
-      ? css`
-          ${activeStyles}
-        `
-      : ""}
-`;
-
-const StyledLink = components.a;
-
-const NavLinkEl = styled(Link)`
-  color: inherit;
-  font-weight: unset;
-  display: block;
-
-  &:hover {
-    color: ${PALETTE.lightGrey};
-  }
-`;
-
-const isInViewport = (elem) => {
-  const bounding = elem.getBoundingClientRect();
-  return (
-    bounding.top >= 0 &&
-    bounding.left >= 0 &&
-    bounding.bottom <=
-      (window.innerHeight || document.documentElement.clientHeight) &&
-    bounding.right <=
-      (window.innerWidth || document.documentElement.clientWidth)
-  );
-};
-
-// eslint-disable-next-line react/prop-types
-const NavItem = ({ isActive, forwardedRef, children, depth }) => {
-  const itemRef = React.useRef();
-  const parentDom = forwardedRef;
-  const { isScrollingDown, setIsNavClicked } = React.useContext(
-    ScrollRouterContext,
-  );
-  const isMobile = useMatchMedia(`(${MEDIA_QUERIES.ltLaptop})`);
-
-  React.useLayoutEffect(() => {
-    if (isActive && parentDom && !isMobile) {
-      const activeItemSize = itemRef.current.getBoundingClientRect();
-
-      /* If the active navigation is not in view
-      For cases when a user scrolled the nav to the point
-      the active nav is out of viewport */
-      if (!isInViewport(itemRef.current)) {
-        itemRef.current.scrollIntoView();
-      }
-
-      /* If scroll direction is down and its active item's top value
-       is bigger than FIXED_NAV_DISTANCE (229px), subtract that amount
-       from scrollTop to keep the consistent top value
-       Its top value gets inconsistent when it hits the separate dropdown category
-      */
-      if (isScrollingDown.current && activeItemSize.top > FIXED_NAV_DISTANCE) {
-        if (activeItemSize.top > FIXED_NAV_DISTANCE) {
-          /* Reset the distance between the active nav and its offset top */
-          parentDom.current.scrollTop +=
-            activeItemSize.top - FIXED_NAV_DISTANCE;
-        } else {
-          parentDom.current.scrollTop += activeItemSize.height;
-        }
-      } else if (
-        !isScrollingDown.current &&
-        activeItemSize.top < FIXED_NAV_DISTANCE
-      ) {
-        if (activeItemSize.top > FIXED_NAV_DISTANCE) {
-          parentDom.current.scrollTop -=
-            activeItemSize.top - FIXED_NAV_DISTANCE;
-        } else {
-          parentDom.current.scrollTop -= activeItemSize.height;
-        }
-      }
-    }
-  }, [isActive, parentDom, isScrollingDown, isMobile]);
-
-  return (
-    <NavItemEl
-      isActive={isActive}
-      depth={depth}
-      ref={itemRef}
-      onClick={() => {
-        setIsNavClicked(true);
-      }}
-    >
-      {children}
-    </NavItemEl>
-  );
-};
```

Comment:
One of the main goals of this PR. This and the other template are much easier to read.